mm/kfence.c

```diff
@@ -363,6 +380,12 @@ bool kfence_handle_page_fault(unsigned long addr)
 	if ((addr < kfence_pool_start) || (addr >= kfence_pool_end))
 		return false;
 
+	if (!READ_ONCE(kfence_enabled)) {
```

Comment:
Why does this need to be here? Should this be an error-recovery path for an assertion/warn failure?

Reply:
If the tool is disabled, we don't want to report any new errors.
However there may be leftover protections causing page faults. If we want the kernel to proceed we need to make those pages accessible.